certbot/certbot/_internal/cli/helpful.py

```diff
@@ -104,12 +104,17 @@ def __init__(self, args, plugins, detect_defaults=False):
         # elements are added by .parse_args()
         self.defaults: Dict[str, Any] = {}
 
+        if "-c" in sys.argv or "--config" in sys.argv:
+            default_config_files = []
+        else:
+            default_config_files = flag_default("config_files")
+
```

Comment:
Please add a test: when `-c/--config` is specified, config values are not read from the default files.

Reply:
I could use some assistance with this.. I've been looking at `cli_test.py` and tried to figure out to add a test for this using `namespace = self.parse([])` and `namespace = self.parse(['-c', tmp_config])` (where `tmp_config` is a `NamedTemporaryFile()`), but it seems `namespace` doesn't contain the default configuration files. It does have a `namespace.config_file`, but that's `None` when no `-c` or `--config` option is used, *not* the default values.

Not sure how to proceed. A simple test asserting `namespace.config_file` is `None` or `tmp_config`, depending on the options provided to `self.parse()` does actually increase the coverage, but that wasn't what you requested..

Reply:
I've added a new test in `helpful_test.py`, as that's the place I can actually check if the default config files are set or not when `-c` or `--config` is used.

However, I think the previously added test in `cli_test.py` isn't *that* bad. Might not be really related to this PR (as this is specifically *not* setting the defaults when using `-c` or `-config`), but I believe it's also not a bad thing to check if `-c` and `--config` actually set `config_file` in the namespace. I believe that check wasn't there before.